src/unity/python/turicreate/test/test_sarray.py

```diff
@@ -3164,4 +3164,27 @@ def test_type_casting(self):
         x['a'] = SArray(x['a'], list)
         self.assertTrue(x['a'].dtype == list)
 
+    def test_filter_by(self):
+        
+        #integer example
+        x = SArray([1,2,3,4,5,6,7])
+        self.assertTrue(np.array_equal(x.filter_by([11,7,2,8,4]), SArray([2,4,7])))
+        self.assertTrue(np.array_equal(x.filter_by([11,7,2,8,4,3], exclude=True), SArray([1,5,6])))
+    
+        #empty SArray
+        self.assertTrue(np.array_equal(x.filter_by([77,22,18,42]), SArray([])))
+        self.assertTrue(np.array_equal(x.filter_by([77,22,18,42], exclude=True).sort(), x))
+        
+        #duplicates
+        self.assertTrue(np.array_equal(x.filter_by([2,2,3,44]),SArray([2,3])))
+        x = SArray([1,2,2,3,4,5,6,7])
+        self.assertTrue(np.array_equal(x.filter_by([2,2,3,44]), SArray([2,2,3])))
+    
+        #strings
+        x = SArray(['dog', 'cat', 'cow', 'horse'])
+        self.assertTrue(np.array_equal(x.filter_by(['cat', 'hamster', 'dog', 'fish', 'bird', 'snake']), SArray(['dog', 'cat'])))
+        self.assertTrue(np.array_equal(x.filter_by(['cat', 'hamster', 'dog', 'fish', 'bird', 'snake'], exclude=True), SArray(['horse', 'cow'])))
+    
+    
```

Comment:
Could you add a test case where you pass a single str value into the filter_by. Something like:
`x.filter_by('dog')`

I'm a little worried about this working properly in Python 3. I'm not sure you want to use `_is_non_string_iterable(...)` to test if it's not an iterable.

Reply:
I think that function is tested in python 3 -- it's used in the SFrame filter_by --  so it should be fine.  However, yes, we should have more test cases.

Reply:
Added a single str test case, and it passes. If you have any other suggestions for test cases please do let me know. So far I have cases for integer, strings, empty return SArrays and duplicates in "values" and the caller SArray.